src/snmalloc/backend_helpers/pagemap.h

```diff
@@ -32,6 +32,24 @@ namespace snmalloc
       std::is_same_v<PagemapEntry, typename ConcreteMap::EntryType>,
       "BasicPagemap's PagemapEntry and ConcreteMap disagree!");
 
+    static_assert(
+      std::is_base_of_v<MetaEntryBase, PagemapEntry>,
+      "BasicPagemap's PagemapEntry type is not a MetaEntryBase");
+
+    /**
+     * This registration allocation hook serves to prevent snmalloc's backend
+     * ranges from consolidating across adjacent OS allocations on platforms
+     * (e.g., Windows or StrictProvenance) where that's required.  It is useful
+     * only for Pagemaps whose Entry type is a subclass of MetaEntryBase; if
+     * you need similar mechanism for different metadata structures, you'll
+     * need a different hook class.
+     */
```

Comment:
Ack; that commentary is stale from an earlier design.  It's the right gist, but refers to things that no longer exist.